src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPCompletionClientConfigFixtureTestCase.java

```diff
@@ -0,0 +1,153 @@
+     *
+     * @param fileName                      the file name
+     * @param fileBodyBefore                the file body before including the embedded typing imperative comment
+     * @param fileBodyAfter                 the file body after the character has been typed and formatting applied
+     * @param mockTextCompletionJson        the mock text/completion JSON response
+     * @param mockCompletionItemResolveJson the mock completionItem/resolve JSON response
+     * @param clientConfigCustomizer        a function that allows the caller to customize the client configuration as
+     *                                      needed for the test scenario
+     */
+    protected void assertTemplateForArguments(@NotNull String fileName,
+                                              @NotNull String fileBodyBefore,
+                                              @NotNull String fileBodyAfter,
+                                              @NotNull String mockTextCompletionJson,
+                                              @NotNull String mockCompletionItemResolveJson,
+                                              @Nullable Consumer<ClientConfigurationSettings> clientConfigCustomizer) {
+        MockLanguageServer.INSTANCE.setTimeToProceedQueries(200);
+
+        CompletionList mockCompletionList = JSONUtils.getLsp4jGson().fromJson(mockTextCompletionJson, CompletionList.class);
+        MockLanguageServer.INSTANCE.setCompletionList(mockCompletionList);
+
+        CompletionItem mockCompletionItem = JSONUtils.getLsp4jGson().fromJson(mockCompletionItemResolveJson, CompletionItem.class);
+        MockLanguageServer.INSTANCE.setCompletionItem(mockCompletionItem);
+
+        Project project = myFixture.getProject();
+        PsiFile file = myFixture.configureByText(fileName, removeCaret(fileBodyBefore));
+        Editor editor = myFixture.getEditor();
+        CaretModel caretModel = editor.getCaretModel();
+
+        // Initialize the language server
+        List<LanguageServerItem> languageServers = new LinkedList<>();
+        try {
+            ContainerUtil.addAllNotNull(languageServers, LanguageServiceAccessor.getInstance(project)
+                    .getLanguageServers(file.getVirtualFile(), null, null)
+                    .get(5000, TimeUnit.MILLISECONDS));
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        // Configure the language server's completion client configuration
+        LanguageServerItem languageServer = ContainerUtil.getFirstItem(languageServers);
+        assertNotNull(languageServer);
+
+        // Use a configurable completion feature
+        LSPClientFeatures clientFeatures = languageServer.getClientFeatures();
+        clientFeatures.setCompletionFeature(new UserDefinedCompletionFeature());
+
+        // Enable completion item resolution
+        languageServer.getServerCapabilities().setCompletionProvider(new CompletionOptions(true, null));
```

Comment:
By default the mock completion provider's options don't enable resolution, so must enable that for this test.